### PR TITLE
Add io.github.snesrev.Zelda3

### PR DESCRIPTION
**Pull request by heroesiiifan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [zelda3](https://github.com/snesrev/zelda3/issues/239)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.snesrev.Zelda3.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+app-id: org.snesrev.Zelda3
```
Comment (hfiguiere) [inline org.snesrev.Zelda3.yml:1]: `snesrev.org` does not exist.

File: org.snesrev.Zelda3.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: zelda3.py
+
+copy-icon: true
```
Comment (hfiguiere) [inline org.snesrev.Zelda3.yml:7]: this does nothing without a rename-icon

File: org.snesrev.Zelda3.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+finish-args:
+ - --device=all
+ - --share=ipc
+ - --socket=x11
+ - --socket=wayland
```
Comment (hfiguiere) [inline org.snesrev.Zelda3.yml:12]: does it support wayland?

File: org.snesrev.Zelda3.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "pyyaml" --no-build-isolation
+   sources:
+    - type: file
+      url: https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5e045f747ff12643df63844/PyYAML-6.0.tar.gz
+      sha256: 68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2
+ 
+ - name: zelda3
+   buildsystem: simple
+   builddir: false
+   no-make-install: true
```
Comment (hfiguiere) [inline org.snesrev.Zelda3.yml:36]:
```suggestion
```

this does nothing with 'simple' build system

File: org.snesrev.Zelda3.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+   build-commands:
+    - pip3 install --verbose --exists-action=i --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} "pyyaml" --no-build-isolation
+   sources:
+    - type: file
+      url: https://files.pythonhosted.org/packages/36/2b/61d51a2c4f25ef062ae3f74576b01638bebad5e045f747ff12643df63844/PyYAML-6.0.tar.gz
+      sha256: 68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2
+ 
+ - name: zelda3
+   buildsystem: simple
+   builddir: false
```
Comment (hfiguiere) [inline org.snesrev.Zelda3.yml:35]:
```suggestion
```

this does nothing with 'simple' build system

**Review by hfiguiere — CHANGES_REQUESTED** [review]
The desktop, icon and appstream file should be submitted upstream

File: org.snesrev.Zelda3.yml (hunk 1 above)
Comment (heroesiiifan) [reply org.snesrev.Zelda3.yml:1]: Would be `com.github.snesrev.Zelda3` okay?

Comment (hfiguiere) [reply org.snesrev.Zelda3.yml:1]: `io.github.snesrev.Zelda3` as per https://docs.flatpak.org/en/latest/conventions.html#application-ids

**Comment by barthalion** [comment]
bot, build io.github.snesrev.Zelda3

**Comment by heroesiiifan** [comment]
bot, build io.github.snesrev.Zelda3

File: io.github.snesrev.Zelda3.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    - sed -i '/(TARGET_EXEC) tables\/zelda3_assets.dat/d' Makefile
+    - make
+   post-install:
+    - install -Dm755 zelda3 ${FLATPAK_DEST}/bin/zelda3
+    - install -Dm755 zelda3.py ${FLATPAK_DEST}/bin/zelda3.py
+    - cp -R . ${FLATPAK_DEST}/bin/src
+    - install -Dm644 io.github.snesrev.Zelda3.appdata.xml -t ${FLATPAK_DEST}/share/metainfo/
+    - install -Dm644 io.github.snesrev.Zelda3.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/io.github.snesrev.Zelda3.svg
+    - |
+        for s in 16 22 24 32 48 64 128 256; do
```
Comment (hfiguiere) [inline io.github.snesrev.Zelda3.yml:43]: you do not need this at all.

Comment (heroesiiifan) [reply io.github.snesrev.Zelda3.yml:43]: On my system, the svg wasn't rendering quite right. that's why I added the pngs.

Comment (heroesiiifan) [reply io.github.snesrev.Zelda3.yml]: done

**Comment by heroesiiifan** [comment]
bot, build io.github.snesrev.Zelda3

**Comment by heroesiiifan** [comment]
bot, build io.github.snesrev.Zelda3

**Pull request closed without merging**